publish-pipeline/content-publish/src/main/scala/org/sunbird/job/content/publish/helpers/ExtractableMimeTypeHelper.scala

```
@@ -0,0 +1,279 @@
+
+  private def generateFileList(sourceFolder: String): List[String] =
+    Files.walk(Paths.get(new File(sourceFolder).getPath)).toArray()
+      .map(path => path.asInstanceOf[Path])
+      .filter(path => Files.isRegularFile(path))
+      .map(path => generateZipEntry(path.toString, sourceFolder)).toList
+
+
+  private def generateZipEntry(file: String, sourceFolder: String): String = file.substring(sourceFolder.length, file.length)
+
+  private def zipIt(zipFile: String, fileList: List[String], sourceFolder: String): Unit = {
+    val buffer = new Array[Byte](1024)
+    var zos: ZipOutputStream = null
+    try {
+      zos = new ZipOutputStream(new FileOutputStream(zipFile))
+      logger.info("Creating Zip File: " + zipFile)
+      fileList.foreach(file => {
+        val ze = new ZipEntry(file)
+        zos.putNextEntry(ze)
+        val in = new FileInputStream(sourceFolder + File.separator + file)
+        try {
+          var len = in.read(buffer)
+          while (len > 0) {
+            zos.write(buffer, 0, len)
+            len = in.read(buffer)
+          }
+        } finally if (in != null) in.close()
+        zos.closeEntry()
+      })
+    } catch {
+      case e: IOException => logger.error("Error! Something Went Wrong While Creating the ZIP File: " + e.getMessage, e)
+    } finally if (zos != null) zos.close()
+  }
+
+  private def extractPackageInCloud(uploadFile: File, obj: ObjectData, extractionType: String, slugFile: Boolean, basePath: String, config: ContentPublishConfig)(implicit cloudStorageUtil: CloudStorageUtil) = {
+    val file = Slug.createSlugFile(uploadFile)
+    val mimeType = obj.mimeType
+    validationForCloudExtraction(file, extractionType, mimeType, config)
+    if (config.extractableMimeTypes.contains(mimeType)) {
+      try {
+        cloudStorageUtil.uploadDirectory(getExtractionPath(obj, config, extractionType), new File(basePath), Option(slugFile))
+      } catch {
+        case _: Exception => println("Remove try catch")
```

Why are we suppressing the exception here. Please remove the try catch statements.